ipatests/test_webui/test_host.py

```diff
@@ -407,11 +435,353 @@ def test_indirect_associations(self):
+                              confirm_btn="cancel")
+        self.wait()
+        fqdn_table = self.find('table[name="fqdn"]', By.CSS_SELECTOR)
+        self.select_record(self.pkey4, parent=fqdn_table)
+        self.button_click('remove')
+        self.dialog_button_click('cancel')
+        self.assert_record(self.pkey4)
+
+        # add duplicate
+        self.add_record(ENTITY, self.data2, negative=True, pre_delete=False)
+        dialog_info = self.get_dialog_info()
+        expected_msg = 'host with name "' + self.data2['pkey'] + \
+                       '" already exist'
+        if expected_msg in dialog_info['text']:
+            self.dialog_button_click('cancel')
+            self.dialog_button_click('cancel')
+        else:
+            assert False, "Duplicate dialog missing or have wrong text."
+
+        # duplicate with pressed keys
+        self.add_record(ENTITY, self.data2, negative=True, pre_delete=False)
+        actions.send_keys(Keys.TAB).perform()
+        self.wait()
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait(3)
+        self.dialog_button_click('cancel')
+        self.assert_no_dialog()
+
+        # remove multiple
+        self.navigate_to_entity(ENTITY)
+        self.select_multiple_records([self.data2, self.data3, self.data4])
+        self.facet_button_click('remove')
+        self.wait()
+        self.check_option('updatedns')
+        self.dialog_button_click('ok')
+        self.assert_notification()
+        self.close_notifications()
+
+        # remove without updatedns
+        self.select_record(self.pkey)
+        self.facet_button_click('remove')
+        self.dialog_button_click('ok')
+        self.assert_notification()
+        self.close_notifications()
+
+    @screenshot
+    def test_negative_add_input(self):
+        """ Test field validations for adding """
+        self.init_app()
+
+        # wrong hostname input
+        hosts_tests = [host.hostname_tilde,
+                       host.hostname_trailing_space,
+                       host.hostname_dash]
+        for hostname_test in hosts_tests:
+            self.add_record(ENTITY, hostname_test, negative=True)
+            dialog_info = self.get_dialog_info()
+            if host.BAD_HOSTNAME_MSG in dialog_info['text']:
+                self.dialog_button_click('cancel')
+                self.dialog_button_click('cancel')
+
+        # leading space in hostname
+        self.add_record(ENTITY, host.hostname_leading_space, negative=True)
+        dialog_info = self.get_dialog_info()
+        if host.BAS_HOSTNAME_SPACE_MSG in dialog_info['text']:
+            self.dialog_button_click('cancel')
+            self.dialog_button_click('cancel')
+
+        # empty hostname
+        self.add_record(ENTITY, host.empty_hostname, negative=True)
+        hostname_i = self.find('.widget[name="hostname"]', By.CSS_SELECTOR)
+        self.assert_field_validation_required(parent=hostname_i)
```

Comment:
To simplify the code, use `self.assert_field_validation_required(field='hostname')` instead of manually constructing selectector. It will be easier to change test if html changes in the future. The same in subsequent usages of `self.assert_field_validation[xxx]()`